src/compiler/path.ts

```diff
@@ -658,7 +662,7 @@ namespace ts {
     //// Path Comparisons
 
     // check path for these segments: '', '.'. '..'
-    const relativePathSegmentRegExp = /(^|\/)\.{0,2}($|\/)/;
+    const relativePathSegmentRegExp = /(?:^|\/)\.\.?(?:$|\/)|\/\//;

```

Comment:
Without this change, the kickout essentially never happens.

Reply:
Non-capturing groups are also cheaper than capturing groups, in general, so should be used any time the capture isn't necessary.